readthedocs/core/resolver.py

```diff
@@ -170,12 +172,10 @@ def resolve(self, project, require_https=False, filename='', private=None,
         ])
         protocol = 'https' if use_https_protocol else 'http'
 
-        return '{protocol}://{domain}{path}'.format(
-            protocol=protocol,
-            domain=domain,
-            path=self.resolve_path(project, filename=filename, private=private,
-                                   **kwargs),
+        path = self.resolve_path(
+            project, filename=filename, private=private, **kwargs
         )
+        return urlunparse((protocol, domain, path, '', query_params, ''))
```

Comment:
:100: on `urlunparse` here!

Reply:
Although, we are using a different nomenclature here than urlparse: `scheme://netloc/path;parameters?query#fragment`.

Also, how is the `#fragment` detected? I suppose that it doesn't comes inside the `query_params` so we will be loosing it here.

Reply:
Fragment are not passed to the server https://stackoverflow.com/questions/2181186/how-to-access-url-hash-fragment-from-a-django-request-object, and on my tests the browser add them automatically to the redirect.

Reply:
Oh, you are right!

I think that I've already commented this in another issue, in fact. Thanks for reminder me this and sorry for creating a confusion here.